src/main/java/net/minecraftforge/common/BiomeManager.java

```diff
@@ -114,6 +126,43 @@ public static void removeBiome(BiomeType type, BiomeEntry entry)
         }
     }
 
+    /**
+     * Adds an OceanBiomeEntry to a type of ocean.
+     * This method should be called during mod loading to add a new OceanBiomeEntry to a type of ocean (e.g. a new Frozen Ocean biome).
+     * This method is threadsafe.
+     *
+     * @param type The type of ocean for this OceanBiomeEntry.
+     * @param entry The OceanBiomeEntry to add.
+     */
+    public static void addOceanBiome(OceanType type, OceanBiomeEntry entry)
+    {
+        synchronized (BiomeManager.class)
+        {
+            forgeOceanBiomes[type.ordinal()].add(entry);
```

Comment:
There should probably be some conflict checking here... What if two people add the same set of biomes but with a different weight? Do we allow that? Throw an exception? Only add the one with the higher weight? Add the weights?

Reply:
It's fine if someone decides to add the same set of biomes with a different weight. 
A. because someone might want ocean biome entries that are the same, but exist for different ocean types, and B. because someone might want two very similar ocean biome entries that have different weights.